netdata/netdata
# Add a chart for the CPU frequency reported by the hardware

##### Summary
`scaling_cur_freq` reports the current CPU frequency as seen by the kernel. It's not the same as the real CPU frequency. We should provide information on the frequency reported by the hardware (`cpuinfo_cur_freq`). The PR adds a new chart `cpu.hw_cpufreq`. It is disabled by default because CPU performance scaling drivers usually read MSR registers to get the information.

Fixes #13116

##### Test Plan
1. Set `hardware cpu frequency = yes` in the `[plugin:proc:/proc/stat]` section of `netdata.conf`.
2. Run Netdata on a Linux system installed on bare metal.
3. Check the `cpu.hw_cpufreq` chart.
